### PR TITLE
Fix bugs in Poco sockets in non-blocking mode, use true non-blocking sockets

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix bugs in Poco sockets in non-blocking mode, use true non-blocking sockets. It fixes possible errors `Cannot read from socket`/`Cannot write to socket`.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Client/Connection.cpp [hunk 1]
```diff
@@ -138,11 +138,6 @@ void Connection::connect(const ConnectionTimeouts & timeouts)
                         socket->impl()->error(err); // Throws an exception
 
                     socket->setBlocking(true);
-
-#if USE_SSL
-                    if (static_cast<bool>(secure))
-                        static_cast<Poco::Net::SecureStreamSocket *>(socket.get())->completeHandshake();
```
Comment (Avogar) [inline src/Client/Connection.cpp:144]: Handshake will be performed lazily

**Comment by robot-clickhouse-ci-1** [comment]
This is an automated comment for commit 42e1e3ae208ed6488b7f30e6e87c88be38b07b17 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/50252/42e1e3ae208ed6488b7f30e6e87c88be38b07b17/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by alexey-milovidov — APPROVED** [review]

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
@Avogar, is it possible to provide a test that will reproduce the issue (even occasionally)?

**Comment by alexey-milovidov** [comment]
This is an urgently needed change, so I've merged it as is, but please make a follow-up with a test, if possible.

**Pull request merged**